### ISSUE TITLE
autorelay: flaky TestWaitForCandidates

### ISSUE THREAD

**Issue by marten-seemann** [libp2p/go-libp2p#1384]
```
=== RUN   TestWaitForCandidates
      autorelay_test.go:176: 
          	Error Trace:	autorelay_test.go:176
          	Error:      	Condition never satisfied
          	Test:       	TestWaitForCandidates
  --- FAIL: TestWaitForCandidates (4.69s)

**Comment by marten-seemann** [comment]
Managed to get a failure with a log trace:
```
=== RUN   TestWaitForCandidates
2022-04-22T17:30:14.004+0100	DEBUG	autorelay	autorelay/relay_finder.go:587	starting relay finder
2022-04-22T17:30:14.156+0100	DEBUG	autorelay	autorelay/relay_finder.go:182	found node	{"id": "QmRUcxbiw2Un1YSoEfrRX5JDWNrDdPmqFHcDcC5fKcVCHf"}
2022-04-22T17:30:14.160+0100	DEBUG	autorelay	autorelay/relay_finder.go:224	node QmRUcxbiw2Un1YSoEfrRX5JDWNrDdPmqFHcDcC5fKcVCHf not accepted as a candidate: doesn't speak circuit v2
2022-04-22T17:30:14.454+0100	DEBUG	autorelay	autorelay/relay_finder.go:182	found node	{"id": "QmYABwYVqF4TUEEGMEUmkveqmFfX3UMS2iqnJjNvRCPF5t"}
2022-04-22T17:30:14.457+0100	DEBUG	autorelay	autorelay/relay_finder.go:232	node supports relay protocol	{"peer": "QmYABwYVqF4TUEEGMEUmkveqmFfX3UMS2iqnJjNvRCPF5t", "supports circuit v2": true}
    autorelay_test.go:199: 
        	Error Trace:	autorelay_test.go:199
        	Error:      	Condition never satisfied
        	Test:       	TestWaitForCandidates
2022-04-22T17:30:17.455+0100	DEBUG	autorelay	autorelay/relay_finder.go:599	stopping relay finder
--- FAIL: TestWaitForCandidates (3.59s)

**Comment by marten-seemann** [comment]
And another, unrelated one:
```
=== RUN   TestWaitForCandidates
2022-04-22T18:37:46.677+0100	DEBUG	autorelay	autorelay/relay_finder.go:587	starting relay finder
2022-04-22T18:37:46.968+0100	DEBUG	autorelay	autorelay/relay_finder.go:182	found node	{"id": "QmWnykjv5AVsRsSxkMayLo4zt3H5tdW4AFNZZPz11bXMoW"}
2022-04-22T18:37:46.970+0100	DEBUG	autorelay	autorelay/relay_finder.go:232	node supports relay protocol	{"peer": "QmWnykjv5AVsRsSxkMayLo4zt3H5tdW4AFNZZPz11bXMoW", "supports circuit v2": true}
2022-04-22T18:37:47.428+0100	DEBUG	autorelay	autorelay/relay_finder.go:182	found node	{"id": "QmdajAAkmZvhA6iitMBVjSYh7hEdvzJ9hcMV6izpTF4ZW2"}
2022-04-22T18:37:47.431+0100	DEBUG	autorelay	autorelay/relay_finder.go:232	node supports relay protocol	{"peer": "QmdajAAkmZvhA6iitMBVjSYh7hEdvzJ9hcMV6izpTF4ZW2", "supports circuit v2": true}
    autorelay_test.go:211: 
        	Error Trace:	autorelay_test.go:211
        	Error:      	Condition never satisfied
        	Test:       	TestWaitForCandidates
2022-04-22T18:38:02.633+0100	DEBUG	autorelay	autorelay/relay_finder.go:599	stopping relay finder
2022-04-22T18:38:02.762+0100	DEBUG	autorelay	autorelay/relay_finder.go:378	adding new relay	{"id": "QmWnykjv5AVsRsSxkMayLo4zt3H5tdW4AFNZZPz11bXMoW"}
--- FAIL: TestWaitForCandidates (3.99s)

**Comment by marten-seemann** [comment]
Fixed by #2120.
